src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2268,6 +2270,30 @@ exprt c_typecheck_baset::do_special_functions(
     typecheck_function_call_arguments(expr);
     return nil_exprt();
   }
+  else if(identifier == CPROVER_PREFIX "is_freeable")
+  {
+    if(expr.arguments().size() != 1)
+    {
+      throw invalid_source_file_exceptiont{
+        CPROVER_PREFIX "is_freeable expects one operand; " +
+          std::to_string(expr.arguments().size()) + " provided",
+        expr.source_location()};
+    }
+    typecheck_function_call_arguments(expr);
+    return nil_exprt();
```

Comment:
What to make of this?

Reply:
I'm using the same approach as the one used with is_fresh : return `nil_exprt()` so that the original expression stays in place. When `do_special_function` returns a non-nil expression it replaces the original expression:

```cpp
  exprt tmp=do_special_functions(expr);

  if(tmp.is_not_nil())
    expr.swap(tmp);
  else
    typecheck_function_call_arguments(expr);
```

Reply:
removing this completely since it seems like declaring the functions in `cprover_builtin_headers.h` and `ansi_c_internal_additions.cpp` suffices to have them type-checked. I don't understand why this was done like this for `is_fresh`